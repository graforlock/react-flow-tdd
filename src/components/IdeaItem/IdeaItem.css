.idea__item {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 12.5%;
  min-width: 100px;
  box-sizing: border-box;
  margin: 7.5px;
  border-radius: 7.5px;
  box-shadow: gainsboro 0 2.5px 7.5px;
}

.idea__item-content {
  position: relative;
}

.idea__item-content::after {
  content: '';
  display: inline-block;
  width: 0; 
  height: 0; 
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  
  border-top: 5px solid white;
  position: absolute;
  left: 0;
  right: 0;
  top: 100%;
  margin: 0 auto;
}

.idea__body {
  margin: 0;
  padding: 20px 5px;
  font-size: 12px;
  color: gray;
}

.idea__header {
  margin: 0;
  padding: 15px 0;
  background: #32557E;
  color: white;
  position: relative;
  border-radius: 7.5px 7.5px 0 0 ;
}

.idea__header h5 {
  margin: 0;
}

.idea__delete-button {
  cursor: pointer;
}

.idea__delete-item {
  padding: 10px 0 0 7.5px;
  height: 30px;
  background: #F0F0F0;
  text-align: left;
}

.idea__delete-text {
  border: none;
  background: transparent;
  color: #CE0B16;
  font-size: 10px;
  font-weight: bold;
  vertical-align: bottom;
}

.idea__delete-icon {
  vertical-align: middle;
  height: 12.5px;
}
