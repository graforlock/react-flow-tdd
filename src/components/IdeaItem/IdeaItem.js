// @flow
import React from 'react'
import { connect } from 'react-redux'
import './IdeaItem.css'
import icon from './trash-icon.svg'

import TextEditor from '../TextEditor'

import { fetchDeleteIdea } from '../../actions'
import type { IdeasFetchAction, IdeaResult } from '../../types'

type Props = {
  idea: IdeaResult,
  fetchDeleteIdea: (ideaId: number) => IdeasFetchAction
}

class IdeaItem extends React.Component<Props> {
  render () {
    const { idea, fetchDeleteIdea } = this.props
    return (
      <li className='idea__item'>
        <section className='idea__item-content'>
          <header className='idea__header'>
            <TextEditor idea={idea} field='TITLE'>
              <h5 className='idea__title'>{idea.title}</h5>
            </TextEditor>
          </header>
          <TextEditor idea={idea} field='BODY'>
            <p className='idea__body'>{idea.body}</p>
          </TextEditor>
        </section>
        <div className='idea__delete-item'>
          <div className='idea__delete-button' onClick={() => fetchDeleteIdea(idea.id)}>
            <img src={icon} className='idea__delete-icon' alt='Delete icon' />
            <button className='idea__delete-text'>Delete</button>
          </div>
        </div>
      </li>
    )
  }
}
export default connect(null, {
  fetchDeleteIdea
})(IdeaItem)
