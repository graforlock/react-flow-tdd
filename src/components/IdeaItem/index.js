import IdeaItem from './IdeaItem'

export default IdeaItem
