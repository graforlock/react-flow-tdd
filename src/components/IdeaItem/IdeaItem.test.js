// @flow
import React from 'react'
import type { ShallowWrapper } from 'enzyme'
import { shallow } from 'enzyme'

import { IdeaItem } from './IdeaItem'

describe('IdeaItem', () => {
  let wrapper: ShallowWrapper
  let mockFetchDeleteIdea
  const idea = {
    id: 1,
    title: 'Title',
    body: '... body ...',
    created_date: new Date('2049')
  }

  beforeEach(() => {
    mockFetchDeleteIdea = jest.fn()
    wrapper = shallow(<IdeaItem idea={idea} fetchDeleteIdea={mockFetchDeleteIdea} />)
  })

  it('renders component and matches current default snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('sends a valid id for idea deletion upon click', () => {
    wrapper.find('.idea__delete-button').simulate('click')

    expect(mockFetchDeleteIdea).toHaveBeenCalledWith(1)
  })
})
