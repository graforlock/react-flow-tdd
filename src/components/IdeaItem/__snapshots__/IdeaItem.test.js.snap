// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaItem renders component and matches current default snapshot 1`] = `
<li
  className="idea__item"
>
  <section
    className="idea__item-content"
  >
    <header
      className="idea__header"
    >
      <Connect(TextEditor)
        field="TITLE"
        idea={
          Object {
            "body": "... body ...",
            "created_date": 2049-01-01T00:00:00.000Z,
            "id": 1,
            "title": "Title",
          }
        }
      >
        <h5
          className="idea__title"
        >
          Title
        </h5>
      </Connect(TextEditor)>
    </header>
    <Connect(TextEditor)
      field="BODY"
      idea={
        Object {
          "body": "... body ...",
          "created_date": 2049-01-01T00:00:00.000Z,
          "id": 1,
          "title": "Title",
        }
      }
    >
      <p
        className="idea__body"
      >
        ... body ...
      </p>
    </Connect(TextEditor)>
  </section>
  <div
    className="idea__delete-item"
  >
    <div
      className="idea__delete-button"
      onClick={[Function]}
    >
      <img
        alt="Delete icon"
        className="idea__delete-icon"
        src="trash-icon.svg"
      />
      <button
        className="idea__delete-text"
      >
        Delete
      </button>
    </div>
  </div>
</li>
`;
